src/components/Filter.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  sortFilter: string
+  setSortFilter: (sortFilter:string) => void;
+};
+
+export const Filter: React.FC<Props> = ({
+  sortFilter,
+  setSortFilter,
+}) => {
+  return (
+    <nav className="filter" data-cy="Filter">
+      <a
+        data-cy="FilterLinkAll"
+        href="#/"
+        className={classNames(
+          'filter__link',
+          { selected: sortFilter === 'all' },
+        )}
+        onClick={() => (
+          sortFilter !== 'all'
+          && setSortFilter('all')
+        )}
```

Comment:
It would be better to make a handler instead of using inline function

Reply:
done

Reply:
I mean, you'd better make something like 
```
onClick={handleClick}
```
in this case,  onClick will fire the same function (handleClick) every time.
If you use an inline function, a new function will be created every time the onClick event takes place